pkg/manifest/a11y.go

```diff
@@ -0,0 +1,407 @@
+package manifest
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// A11y holds the accessibility metadata of a Publication.
+//
+// https://www.w3.org/2021/a11y-discov-vocab/latest/
+// https://readium.org/webpub-manifest/schema/a11y.schema.json
+type A11y struct {
+	ConformsTo            []A11yProfile             `json:"conformsTo,omitempty"`           // An established standard to which the described resource conforms.
+	Certification         *A11yCertification        `json:"certification,omitempty"`        // Certification of accessible publications.
+	Summary               string                    `json:"summary,omitempty"`              // A human-readable summary of specific accessibility features or deficiencies, consistent with the other accessibility metadata but expressing subtleties such as "short descriptions are present but long descriptions will be needed for non-visual users" or "short descriptions are present and no long descriptions are needed."
+	AccessModes           []A11yAccessMode          `json:"accessMode,omitempty"`           // The human sensory perceptual system or cognitive faculty through which a person may process or perceive information.
+	AccessModesSufficient [][]A11yPrimaryAccessMode `json:"accessModeSufficient,omitempty"` //  A list of single or combined accessModes that are sufficient to understand all the intellectual content of a resource.
+	Features              []A11yFeature             `json:"feature,omitempty"`              // Content features of the resource, such as accessible media, alternatives and supported enhancements for accessibility.
+	Hazards               []A11yHazard              `json:"hazard,omitempty"`               // A characteristic of the described resource that is physiologically dangerous to some users.
+}
+
+// NewA11y creates a new empty A11y.
+func NewA11y() A11y {
+	return A11y{
+		ConformsTo:            []A11yProfile{},
+		AccessModes:           []A11yAccessMode{},
+		AccessModesSufficient: [][]A11yPrimaryAccessMode{},
+		Features:              []A11yFeature{},
+		Hazards:               []A11yHazard{},
+	}
+}
+
+func (a A11y) IsEmpty() bool {
```

Comment:
This is a good approach, it might be a good idea to replicate this in the other structs in this repo